--- a/src/util.cpp
+++ b/src/util.cpp
@@ -19,6 +19,7 @@
 #include "utiltime.h"
 
 #include <chrono>
+#include <type_traits>
 #include <stdarg.h>
 
 #include <boost/date_time/posix_time/posix_time.hpp>
@@ -477,13 +478,32 @@ void ClearDatadirCache()
     pathCachedNetSpecific = boost::filesystem::path();
 }
 
+template<typename T>
+typename std::enable_if<std::is_integral<T>::value, std::time_t>::type
+BoostFilesystemLastWriteTimeImpl(T t)
+{
+    return static_cast<std::time_t>(t);
+}
+
+template<typename T>
+typename std::enable_if<!std::is_integral<T>::value, std::time_t>::type
+BoostFilesystemLastWriteTimeImpl(T ft)
+{
+    using namespace std::chrono;
+    return system_clock::to_time_t(time_point_cast<system_clock::duration>(
+        ft - T::clock::now() + system_clock::now()));
+}
+
 std::time_t BoostFilesystemLastWriteTime(const boost::filesystem::path& path)
 {
-    const boost::filesystem::file_time_type ft = boost::filesystem::last_write_time(path);
-    using namespace std::chrono;
-    return system_clock::to_time_t(time_point_cast<system_clock::duration>(
-        ft - boost::filesystem::file_time_type::clock::now() + system_clock::now()));
+#if BOOST_VERSION < 106600
+    return boost::filesystem::last_write_time(path);
+#else
+    return BoostFilesystemLastWriteTimeImpl(boost::filesystem::last_write_time(path));
+#endif
 }
 
 boost::filesystem::path GetConfigFile()
